query/src/sql/sql_parser.rs

```diff
@@ -223,6 +224,34 @@ impl<'a> DfParser<'a> {
         Ok(DfStatement::Explain(explain_plan))
     }
 
+    // parse show databases where database = xxx or where database
+    fn parse_show_databases(&mut self) -> Result<DfStatement, ParserError> {
+        match self.parser.parse_keyword(Keyword::WHERE) {
+            true => {
+                if !self.consume_token("DATABASE") {
```

Comment:
Seems we don't need the keyword `DATABASE`.

1. `WHERE` -->  parser.parse_expr()   -> `append  format!("where {}", expr)`
2.  `like` --> parser.peek_token --> `append format!( "where name like {}") ` to the sql

```

Reply:
https://dev.mysql.com/doc/refman/8.0/en/show-databases.html

Reply:
in the #1390: 
> SHOW DATABASES WHERE `Database` = 'database_1';
SHOW DATABASES WHERE `Database` like '%database_1%';

if I  understand correctly,   we should support all these show database statements?
> SHOW DATABASES WHERE `Database` = 'database_1';
SHOW DATABASES WHERE `Database` like '%database_1%';
SHOW DATABASES WHERE expr;
SHOW DATABASES LIKE 'pattern';

Reply:
Yes, maybe use parse_expr is better : )
``` rust
let selection = if self.parser.parse_keyword(Keyword::WHERE) {
    Some(self.parser.parse_expr()?)
} else {
    None
};
```

Reply:
Ok, thanks for suggestion.